packages/launch/src/focused-launch/plan-details/index.tsx

```diff
@@ -3,22 +3,80 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import { Link } from 'react-router-dom';
+import * as React from 'react';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import { Plans } from '@automattic/data-stores';
+import PlansGrid from '@automattic/plans-grid';
+import { Title, SubTitle } from '@automattic/onboarding';
+import { useHistory } from 'react-router-dom';
 
 /**
  * Internal dependencies
  */
-import { Route } from '../route';
+import { LAUNCH_STORE } from '../../stores';
+import GoBackButton from '../go-back-button';
 
 import './style.scss';
 
 const PlanDetails: React.FunctionComponent = () => {
+	const domain = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedDomain() );
+	const selectedPlan = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedPlan() );
+	const history = useHistory();
+
+	const { updatePlan } = useDispatch( LAUNCH_STORE );
+
+	const hasPaidDomain = domain && ! domain.is_free;
```

Comment:
If `domain` is `null` or `undefined` you will get these values assigned to `hasPaidDomain` and it'll be coerced into a falsely value.

If you changed `const hasPaidDomain = domain && ! domain.is_free;` to `const hasPaidDomain = domain?.is_free === false;` then you'll get a more predictable boolean returned.

Not a blocker though, as it'll be coerced into a falsely value anyway.

Reply:
Great idea but Typescript won't like because `is_free` is typed wrong.